### ISSUE TITLE
TR3 save is loaded and game quits to desktop

### ISSUE THREAD

**Issue by StenApp** [JulianOzelRose/TRR-SaveMaster#2]
I used version 0.8 to modify the savestate 57 TR3, number of medi paks, activated deagle, shotgun and grenade launcher plus gave them more ammunition

[TR3 save 57.rar.txt](https://github.com/JulianOzelRose/TRR-SaveMaster/files/14320037/TR3.save.57.rar.txt)

**Comment by JulianOzelRose** [comment]
I repaired your savegame, along with the changes you were trying to apply -- it's attached below. It looks like the editor somehow got the ammo index wrong, and overwrote part of the index array. When I replaced the array with what it should have been, it stopped crashing. Then I applied the changes you were trying to, and it worked too.

Almost seems like a fluke since it worked for me. If I had to guess, the game state (and offsets) changed from the time you selected it in the editor to the time you clicked "Save" in the editor -- so it used the wrong offsets. I patched the editor to recalculate the offsets before writing for TR3, so hopefully that should prevent this issue in the future. [This binary](https://github.com/JulianOzelRose/TRR-SaveMaster/tree/master/TRR-SaveMaster/bin/x64/Release) contains the change.

[savegame.zip](https://github.com/JulianOzelRose/TRR-SaveMaster/files/14327291/savegame.zip)
